src/charm.py

```diff
@@ -184,6 +214,15 @@ def _on_database_changed(self, event: DatabaseCreatedEvent) -> None:
         self.unit.status = MaintenanceStatus("Retrieving database details")
         self._update_container(event)
 
+    def _on_ingress_ready(self, event: IngressPerAppReadyEvent) -> None:
+        if self.unit.is_leader():
```

Comment:
> Currently there is a problem with the traefik charm and all traffic will be redirected to the Kratos leader.

Will it all be handled on traefik side? no action needed on kratos?

Reply:
Yes, I think that no changes (or very little) will be needed on the Kratos side to fix this.